cli/src/cli_util.rs

```diff
@@ -3940,8 +3949,12 @@ impl CliRunner {
         let command_helper = CommandHelper {
             data: Rc::new(command_helper_data),
         };
-        for start_hook_fn in self.start_hook_fns {
-            start_hook_fn(ui, &command_helper)?;
+        for dispatch_hook_fn in self.dispatch_hook_fn {
+            let old_dispatch_fn = self.dispatch_fn;
+            let new_dispatch_fn = move |ui: &mut Ui, command_helper: &CommandHelper| {
+                dispatch_hook_fn(ui, command_helper, old_dispatch_fn)
+            };
+            self.dispatch_fn = Box::new(new_dispatch_fn);
```

Comment:
nit: I think it's simpler to move `self.dispatch_fn` to local variable. We can also use `.into_iter().fold()`.

Reply:
Done